NetworkConnection connection
NetworkDevice device
uint8 connection_state
string ipv4_address

uint8 ACTIVATED_STATE=0
uint8 ACTIVATING_STATE=1
uint8 DEACTIVATING_STATE=2
uint8 UNKNOWN_STATE=3